Updating Pairwise Test Instructions

Dbarnes1 opened the pull request:
This includes commits that revert the faulty legacy Military Overlay information models, but the updates to the readme for pairwise test data and comprehensive test data is the focus of this pull request. Aka, only focus on the last 3 commits in this PR

csmoore commented:
Good deal - looks like all of this stuff has really progressed and is much more repeatable now.  

We don't expect just anyone to be able to do these more advanced/obscure use cases such as generating test datasets, but certainly the 3 of us (or someone with our background/skills) should be able to do, and I think we are at that point now.

csmoore commented on data/mil2525d/test_data/README.md:
```diff
@@ -14,10 +14,15 @@ See the Component Sections for more information:
+
 	
-## Stepts to create a Comprehensive Test Dataset
+## Stepts to create a Comprehensive Test Dataset (Manual)
```
"Stepts"

Dbarnes1 commented:
@csmoore updated this PR with the fixed layer packages for 2525C, Bc2, and D. In addition, I added the new version of 2525D Pairwise and fixed a typo from the readme that you noticed. Let me know what you think.

csmoore commented on data/mil2525c_b2/test_data/Complete_test_scripts/README.md:
```diff
@@ -0,0 +1,17 @@
+
+* Connect the `Complete_test_scripts` folder to one of the Military Overlay Information Models for ArcGIS Pro.
+* Open the `FilterCSVMilSymbols` Script. For the File to Filter parameter, use any 2525 source .csv, and the script will create output .csv files filtered by their appendix. For example, the source .csv files for 2525 C and Bc2 can be found [here](https://github.com/Esri/military-features-data/tree/v.next/data/mil2525c_b2/test_data/truth_data). 
+* Choose an appropriate output folder and run the script. 
```
You normally want to use a relative link ex: `"./truth_data"` so the link still works if cloned/changed - may not matter in this case if that folder/link never changes

csmoore commented:
@Dbarnes1 - looking at this now - do you have the truth data that was used for the 2525D comprehensive data? Can/should we include that here?

csmoore commented on data/mil2525d/test_data/README.md:
```diff
@@ -14,10 +14,15 @@ See the Component Sections for more information:
 	* Note: ArcGIS Professional required
+
+## Scripts and Models for Creating Comprehensive Test Datasets
+* This [link](https://github.com/Esri/military-features-data/tree/v.next/data/mil2525c_b2/test_data/Complete_test_scripts) will take you to the location of a script and a model that automates the manual process outlined below. 
```
Change to relative path/link

csmoore commented on data/mil2525d/utilities/pairwise-testing/README.md:
```diff
@@ -3,7 +3,18 @@
+* Navigate (`cd`) to the location of the input file you want to run. 
+* Enter `InputFolder> pict InputFile.txt > OutputFile.xls` to run the PICT application. You should see an `OutputFile.xls` appear in the same folder as your input.
+* Save each output .xls file as a .csv in a new folder. Make sure your CSV has a OID primary key. 
+* Open up ArcGIS Desktop and add this [toolbox](https://github.com/Esri/military-features-data/blob/v.next/data/mil2525c_b2/test_data/Complete_test_scripts/CompleteTestData.tbx) to your project.
```
Change the 2 links (line 12, 17) to relative path/link

csmoore commented:
@Dbarnes1 - some other thoughts looking at this PR:

1. Should we remove the non-schema only versions or do the same workaround for them?
2. Add the workaround`*` somewhere - like the [Known Issues Wiki](https://github.com/Esri/military-features-data/wiki/Known-Issues)
3. Are both pairwise datasets still needed? PairwiseTestProject.ppkx & RessurectPairwise.ppkx (not the best name for this one)

`*` Spatial Index Packaging Workaround:
```
•  Have your data ready to package
•  Immediately before packaging:
o   Run Remove Spatial Index
o   Run Add Spatial Index
  This should set grids to 0,0,0
•  Now package.
```

csmoore commented:
I'm not able to add the 2525D lpk to the map on Pro 1.2:  `mil2525d\overlay_templates\layerpackages\MilitaryOverlay.lpkx` - was this created on 1.3 or something (I noticed that the 1.3 ppk's do load on 1.2, but perhaps the lpks don't)?

csmoore commented:
I do notice the ppks were created with 1.3 - I was hoping to update the 1.2 Release with this PR (so the lpks would work in the release)

![image](https://cloud.githubusercontent.com/assets/3090809/17439398/e4bc87bc-5af5-11e6-934a-9c447b534b00.png)

csmoore commented:
DCompleteTestData_PointsOnly.ppkx looks good.

Only suggestion is for the test data "SymbolEntity" might have been a more descriptive field name than "Value" (and is the name we usually use in military info schema)

I never noticed the "?" icon when entity = "000000" (but we may not have tested this value before) - I guess this behavior is OK/correct/acceptable

![image](https://cloud.githubusercontent.com/assets/3090809/17439567/b7d53194-5af6-11e6-9726-5a3d4135bdba.png)

csmoore commented:
Done looking at this - various minor comments above - the only major item is decide if this PR will target 1.2 or 1.3. I was hoping for 1.2 so I could update the broken lpks in the [current/latest release](https://github.com/Esri/military-features-data/releases/tag/1.2.1) - but I don't think we ever discussed which version this PR targets.

Dbarnes1 commented:
@csmoore updated the PR with these commits to reflect your review.

Specifically:
1. Got rid of non schema only layer packages
2. added a 2525D layer package for pro 1.2
3. fixed doc with relative paths
4. added 2525D truth data
5. got rid of old pairwise test dataset project for 2525D and renamed current one to simply "pairwise"

I'll update the lpkx packaging workaround in the readme

csmoore commented on data/mil2525c_b2/test_data/Complete_test_scripts/README.md:
```diff
@@ -0,0 +1,17 @@
+## Instructions 
+
+* Connect the `Complete_test_scripts` folder to one of the Military Overlay Information Models for ArcGIS Pro.
+* Open the `FilterCSVMilSymbols` Script. For the File to Filter parameter, use any 2525 source .csv, and the script will create output .csv files filtered by their appendix. For example, the source .csv files for 2525 C and Bc2 can be found [here](./truth_data). 
```
Close - it should be `../truth_data` (up one folder)

csmoore commented on data/mil2525d/test_data/README.md:
```diff
@@ -14,10 +14,15 @@ See the Component Sections for more information:
 	* Note: ArcGIS Professional required
+
+## Scripts and Models for Creating Comprehensive Test Datasets
+* This [link](./mil2525c_b2/test_data/Complete_test_scripts) will take you to the location of a script and a model that automates the manual process outlined below. 
```
Bad link - probably `../../mil2525c_b2` (up 2 folders...)

csmoore commented on data/mil2525d/utilities/pairwise-testing/README.md:
```diff
@@ -3,7 +3,18 @@
+* Navigate (`cd`) to the location of the input file you want to run. 
+* Enter `InputFolder> pict InputFile.txt > OutputFile.xls` to run the PICT application. You should see an `OutputFile.xls` appear in the same folder as your input.
+* Save each output .xls file as a .csv in a new folder. Make sure your CSV has a OID primary key. 
+* Open up ArcGIS Desktop and add this [toolbox](./mil2525c_b2/test_data/Complete_test_scripts/CompleteTestData.tbx) to your project.
```
bad link

csmoore commented:
Which file is the 2525D layer package for 1.2? I tried both MilitaryOverlay1.0.0.lpkx & MilitaryOverlay2.0.0.lpkx and neither seemed to work.

Also we probably don't want to have multiple versions in the GitHub/filenames (this is the first I have seen this) - things could quickly get out of hand - only the most current version (for 1.2 in this one case) should be in GitHub.

Dbarnes1 commented:
@csmoore updated with relative path fixes and fixes to MilitaryOverlay.lpkx

Dbarnes1 commented:
@csmoore added a toolbox that contains the Model I've been using to create a new spatial index before packaging the layer packages. I added a new folder "layer-package-utilities" under 2525D utilities, which is where the toolbox is located.

csmoore commented:
@Dbarnes1 Very nice - way to put these skills to work - question: shouldn't the model first "remove the spatial index" or do I not understand the [workaround completely](https://github.com/Esri/military-features-data/pull/231#issuecomment-237849371) ?

I think we should still/also include this in the ["Known Issues" wiki](https://github.com/Esri/military-features-data/wiki/Known-Issues)

![image](https://cloud.githubusercontent.com/assets/3090809/17489395/c3d66e8c-5d6b-11e6-894e-c623708f2eb8.png)

Dbarnes1 commented:
@csmoore I've been testing this out, and it seems that when it adds the spatial index, it is resetting the parameters to 0,0,0 so I don't think the remote spatial index step is mandatory. Regardless of whether the feature class had a spatial index or not, adding a new one is what seems to be resetting it and making them work again.

csmoore commented:
Good deal - thanks for sticking with this and figuring out this problem+solution ("invalid spatial index" on lpk unpack).

I'll merge&update the Pro 1.2 release and then you can resume/start doing things in 1.3 in v.next.
